packages/phoenix-event-display/src/managers/three-manager/import-manager.ts

```diff
@@ -392,6 +398,7 @@ export class ImportManager {
             shininess: 0,
             side: side,
           });
+          if (opacity) child['material'].opacity = opacity;
```

Comment:
We can get the opacity directly from the geometry right here instead of calculating it in the calling function.

```ts
child.material = new MeshPhongMaterial({
  color: color,
  shininess: 0,
  side: side,
  opacity: geometry.userData.opacity,
});
```

If `geometry.userData.opacity` is `undefined` (when no opacity is set in `userData`), three.js will automatically use a value of `1.0`. But if we want to be explicit about the fallback value, we can also use nullish coalescing `opacity: geometry.userData.opacity ?? 1`.

Reply:
:+1:  will change to your version tomorrow

Reply:
Ready to go. And on top the change is now a single line ! thanks for the very useful review. I was making it way more complicated than needed !

Reply:
It seems that the change hasn't been pushed yet.

Reply:
oh, I pushed to the wrong repo ! Fixed